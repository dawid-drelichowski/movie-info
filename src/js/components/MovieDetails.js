import React, {Component} from 'react'
import PropTypes from 'prop-types'

export default class MovieDetails extends Component {
  static propTypes = {
    config: PropTypes.object.isRequired,
    details: PropTypes.object.isRequired
  }
  render () {
    const config = this.props.config.movieDb.images,
      details = this.props.details
    return <article>
      <dl>
        <dd>Rating:</dd>
        <dt>{details.vote_average} ({details.vote_count} votes)</dt>
        <dd>Genres:</dd>
        <dt>{details.genres.map(genre => genre.name).join(', ')}</dt>
        <dd>Release date:</dd>
        <dt>{details.release_date}</dt>
      </dl>
      {details.poster_path ? <img src={`${config.url}${config.size.big}${details.poster_path}`}/> : ''}
      <p>{details.overview}</p>
    </article>
  }
}
