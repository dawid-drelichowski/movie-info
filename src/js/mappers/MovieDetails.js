import Abstract from 'mappers/Abstract'
import Details from 'models/MovieDetails'
import GenreMapper from 'mappers/MovieGenre'

export default class extends Abstract {
  static mapping = new Map([
    ['id', 'id'],
    ['title', 'title'],
    ['imagePath', 'poster_path'],
    ['averageVote', 'vote_average'],
    ['voteCount', 'vote_count'],
    ['releaseDate', 'release_date'],
    ['overview', 'overview']
  ])
  static map (data) {
    const model = new Details()
    if (data.genres.length) {
      model.genres = GenreMapper.mapCollection(data.genres)
    }
    this.performMapping(data, model)
    return model
  }
}
