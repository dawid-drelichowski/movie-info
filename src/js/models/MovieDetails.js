export default class {
  id = 0
  title = ''
  imagePath = ''
  averageVote = 0.0
  voteCount = 0
  genres = []
  releaseDate = ''
  overview = 0
}
