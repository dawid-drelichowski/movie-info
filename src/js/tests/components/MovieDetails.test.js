import React from 'react'
import MovieDetails from 'components/MovieDetails'
import renderer from 'react-test-renderer'
import config from 'tests/mocks/config.json'
import movie from 'tests/mocks/movie.json'

describe('Movie details component', () => {
  function getRenderedComponent (details) {
    return renderer.create(
      <MovieDetails config={config} details={details} />
    )
  }

  it('shouldn\'t have image', () => {
    expect(getRenderedComponent({...movie, poster_path: undefined}).toJSON()).toMatchSnapshot()
  })

  it('should have image', () => {
    expect(getRenderedComponent(movie).toJSON()).toMatchSnapshot()
  })
})
