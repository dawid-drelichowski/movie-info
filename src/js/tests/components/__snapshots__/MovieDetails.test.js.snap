// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie details component should have image 1`] = `
<article>
  <dl>
    <dd>
      Rating:
    </dd>
    <dt>
      6.4
       (
      4412
       votes)
    </dt>
    <dd>
      Genres:
    </dd>
    <dt>
      Family, Animation, Adventure, Comedy
    </dt>
    <dd>
      Release date:
    </dd>
    <dt>
      2015-06-17
    </dt>
  </dl>
  <img
    src="https://image.mock.api/big/q0R4crx2SehcEEQEkYObktdeFy.jpg"
  />
  <p>
    Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
  </p>
</article>
`;

exports[`Movie details component shouldn't have image 1`] = `
<article>
  <dl>
    <dd>
      Rating:
    </dd>
    <dt>
      6.4
       (
      4412
       votes)
    </dt>
    <dd>
      Genres:
    </dd>
    <dt>
      Family, Animation, Adventure, Comedy
    </dt>
    <dd>
      Release date:
    </dd>
    <dt>
      2015-06-17
    </dt>
  </dl>
  
  <p>
    Minions Stuart, Kevin and Bob are recruited by Scarlet Overkill, a super-villain who, alongside her inventor husband Herb, hatches a plot to take over the world.
  </p>
</article>
`;
